Update security docs to remove outdated "enforcing TLS" related sections (#4796)
boto/boto3: docs/source/guide/security.rst

Reviewer: Is this still true?  

Our minimum version of urllib3 is now 2.x, which has 1.2 as [the minimum by default](https://urllib3.readthedocs.io/en/stable/v2-migration-guide.html#https-requires-tls-1-2).  What scenario would a user still see lower than 1.1?

Author: I see. I will update the first sentence to reflect that.

Enforcing TLS 1.2
------------------

.. note::
    If you are using urllib3 2.x or later (the default for new installations), TLS 1.2 is already enforced by default and no additional steps are needed.
    You can verify your urllib3 version by running ``python -c "import urllib3; print(urllib3.__version__)"``.

>     The steps below are only necessary if you are using urllib3 1.x or need to enforce TLS 1.3

Reviewer: ```suggestion
    The steps below are only necessary if you are using urllib3 1.x and need to enforce TLS 1.2
```

Below your instructions are still using `no-tls1 no-tls1_1 no-shared` to disable 1.0 and 1.1.  1.2 is still supported.  Though personally, I think I'd rewrite this below, and edit this bit: 
> To ensure the SDK or CLI does not negotiate for anything earlier than TLS 1.2, you need to recompile OpenSSL and Python. First copy the following content to create a script and run it::

->

> As of urllib3 2.x, the default minimum SSL version for all calls through the SDK and CLI is 1.2.  For users who are still using older versions of urllib3, additional steps are necessary to ensure the SDK or CLI does not negotiate for anything earlier than TLS 1.2.  You need to recompile OpenSSL and Python. First copy the following content to create a script and run it:

That's off the cuff, I'd have to edit it for a while before I got something I was happy with.

Compile OpenSSL and Python
~~~~~~~~~~~~~~~~~~~~~~~~~~~

As of urllib3 2.x, the default minimum SSL version for all calls through the SDK and CLI is 1.2.
For users who are still using older versions of urllib3, additional steps are necessary to ensure the SDK or CLI does not negotiate for anything earlier than TLS 1.2.

> You need to recompile OpenSSL and Python. First copy the following content to create a script and run it::

Reviewer: Sorry, when I wrote that out it wasn't well written or high enough quality to be accepted; it was showing the thought process, not suggesting final wording.  Here's some better wording:  

```suggestion
The SDK and CLI enforce a minimum TLS version of 1.2 for all requests when running on urllib3 2.x or later. If you're still on urllib3 1.x and unable to upgrade, you'll need to take additional steps to prevent the SDK or CLI from negotiating anything older than TLS 1.2.

This requires rebuilding OpenSSL and Python from source. Save the following script 
and run it::
```

Still though, my opinion here is that this doc is solving a problem for an audience that has mostly disappeared.  We'd be better off reconsidering if this section belongs here at all; it's six years old, and a lot has changed in that time.  The majority of boto/cli users are likely already on urllib3 2.x; for those who are not, it's easier for them to upgrade than it is to follow the instructions written here.

    #!/usr/bin/env bash
    set -e

    OPENSSL_VERSION="3.5.0"
    OPENSSL_PREFIX="/opt/openssl-with-min-tls1_2"

>     PYTHON_VERSION="3.10.0"

Reviewer: Should we not be using more recent versions for both Python and OpenSSL here?

Author: `Python 3.10.0` is the minimum supported version for boto3. `OpenSSL 3.5.0` is the [current LTS release, supported until 2030](https://openssl-library.org/post/2025-02-20-openssl-3.5-lts/). Using the minimum Python version ensures the instructions work for all supported customers, and the OpenSSL LTS gives the longest window before this goes stale again, so I decided to use these versions.

Reviewer: Yeah, I don't know if we should be using the minimum?  I'd at least want the most recent version to include security releases
```suggestion
    PYTHON_VERSION="3.10.20"
```

Same with openssl, should be 3.5.7 (still long term support, but is a higher patch release).
